docs/MostUsedConditions.md

```diff
@@ -0,0 +1,259 @@
+
+#### RemoveDuplicateBooleanNot
+This processor removes duplicate negations. So if you have an complex condition, and split it off, it could be possible in the printed condition is something like `!!$user`. You can achieve this with [this](examples/MostUsedConditions/removeduplicatebooleannotprocessor.php) source-code and the processors [SplitLogicalOperator](#splitlogicaloperator) and [NegateBooleanNot](#negatebooleannot) together:
+```php
+<?php
+$user = null;
+if (!(30 === date('d') && !(null === $user))) {
+    // do something
+}
+```
+If you add the RemoveDuplicateBooleanNot-Processor, it will print `null === $user` and removes the `!!`.
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=SplitLogicalOperator,RemoveDuplicateBooleanNot --include-files=removeduplicatebooleannotprocessor.php`
+
+#### RemoveSingleFullyQualifiedName
+This processor will ignore backslashes from fully qualified name functions or class name (e.g. new \DateTime), because maybe you don't use them or import an class. Codestyle can change often. See [this](examples/MostUsedConditions/removesinglefullyqualifiednameprocessor.php) source-code:
+```php
+<?php
+if (strtolower('Chris') === 'chris') {
+    // hey chris
+}
+
+if (\strtolower('Chris') === 'chris') {
+    // chris again :o
+}
+```
+The RemoveSingleFullyQualifiedName-Processor will count the condition as 2, without the processor it will count both as single condition.
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=RemoveSingleFullyQualifiedName --include-files=removesinglefullyqualifiednameprocessor.php`
+
+<img src="./images/MostUsedConditions/removesinglefullyqualifiednameprocessor.png" height="140">
+
+#### RemoveCast
+This processor will any (cast)-casts (like `(int)`). I saw this in some legacy applications. See [this](examples/MostUsedConditions/removecastprocessor.php) source-code:
+```php
+<?php
+
+$someReturnValue = '123';
+if ((int) $foo === 123) {
+    // found numeric 123
+}
+```
+The RemoveCast-Processor will count the condition as 1, but without (int).
+
+`php bin/php-analyzer most-used-conditions --without-affected-by-processors --visitors=If --processors=RemoveCast --include-files=removecastprocessor.php`
+
+<img src="./images/MostUsedConditions/removecastprocessor.png" height="140">
+
+### Flip-Check
+- with `--with-flip-check` the command will try to swap both sides of `==`, `!=`, `===` and `!==` and check if it already exists. If yes it will mark it with an `flipped`-flag. See [this](examples/MostUsedConditions/flipcheck.php) source-code:
+```php
+<?php
+if(date('Y') === 2019) {
+    echo '2019!!!';
+}
+
+if (2019 === date('Y')) {
+    echo '2019!!!';
+}
+```
+Normally the command will count both conditions as found once, with the option `--with-flip-check` it will be count as twice and flip the second condition.
```

Comment:
```suggestion
Normally the command will count both conditions as found once, with the option `--with-flip-check` it will be counted twice and flip the second condition.
```